core/payment-driver/zksync/src/zksync/wallet.rs

```diff
@@ -4,75 +4,83 @@
 
 // External crates
 use bigdecimal::{BigDecimal, Zero};
-use futures3::TryFutureExt;
-use lazy_static::lazy_static;
 use num::BigUint;
 use std::env;
 use std::str::FromStr;
 use zksync::operations::SyncTransactionHandle;
 use zksync::types::BlockStatus;
 use zksync::zksync_types::{tx::TxHash, Address, TxFeeTypes};
-use zksync::{Network, Provider, Wallet, WalletCredentials};
+use zksync::{Network as ZkNetwork, Provider, Wallet, WalletCredentials};
 use zksync_eth_signer::EthereumSigner;
 
 // Workspace uses
-use ya_payment_driver::model::{AccountMode, Exit, GenericError, Init, PaymentDetails};
+use ya_payment_driver::{
+    db::models::Network,
+    model::{AccountMode, Exit, GenericError, Init, PaymentDetails},
+};
 
 // Local uses
 use crate::{
+    network::get_network_token,
     zksync::{faucet, signer::YagnaEthSigner, utils},
-    ZKSYNC_TOKEN_NAME,
+    DEFAULT_NETWORK, ZKSYNC_TOKEN_NAME,
 };
 
-lazy_static! {
-    pub static ref NETWORK: Network = {
-        let chain_id = env::var("ZKSYNC_CHAIN")
-            .unwrap_or("rinkeby".to_string())
-            .to_lowercase();
-        match chain_id.parse() {
-            Ok(network) => network,
-            Err(_) => panic!(format!("Invalid chain id: {}", chain_id)),
-        }
-    };
-    static ref PROVIDER: Provider = match env::var("ZKSYNC_RPC_ADDRESS").ok() {
-        Some(rpc_addr) => Provider::from_addr(rpc_addr),
-        None => Provider::new(*NETWORK),
-    };
-}
-
-pub async fn account_balance(address: &str) -> Result<BigDecimal, GenericError> {
+pub async fn account_balance(address: &str, network: Network) -> Result<BigDecimal, GenericError> {
     let pub_address = Address::from_str(&address[2..]).map_err(GenericError::new)?;
-    let acc_info = get_provider()
+    let acc_info = get_provider(network)
         .account_info(pub_address)
         .await
         .map_err(GenericError::new)?;
-    let balance_com = acc_info
+    // TODO: implement tokens, replace None
+    let token = get_network_token(network, None);
+    let mut balance_com = acc_info
         .committed
         .balances
-        .get(ZKSYNC_TOKEN_NAME)
+        .get(&token)
         .map(|x| x.0.clone())
         .unwrap_or(BigUint::zero());
+    // Hack to get GNT balance for backwards compatability
+    // TODO: Remove this if {} and the `mut` from `let mut balance_com`
+    if network == Network::Rinkeby && balance_com == BigUint::zero() {
```

Comment:
Please contact Matter Labs team regarding this issue.

Reply:
this one is the "known" issue, balances returns "GNT" still, will reach out to them about transfer failing to work with tGLM